pkg/gardenlet/controller/backupentry/reconciler_test.go

```diff
@@ -186,83 +188,247 @@ var _ = Describe("Controller", func() {
+			Expect(gardenClient.Create(ctx, workloadIdentity)).To(Succeed())
+			extensionSecret.Annotations = map[string]string{
+				"gardener.cloud/timestamp":                                now.Format(time.RFC3339Nano),
+				"workloadidentity.security.gardener.cloud/context-object": `{"kind":"BackupEntry","apiVersion":"core.gardener.cloud/v1beta1","name":"bar","namespace":"test","uid":""}`,
+				"workloadidentity.security.gardener.cloud/name":           workloadIdentity.Name,
+				"workloadidentity.security.gardener.cloud/namespace":      workloadIdentity.Namespace,
+			}
+			extensionSecret.Labels = map[string]string{
+				"security.gardener.cloud/purpose":                   "workload-identity-token-requestor",
+				"workloadidentity.security.gardener.cloud/provider": "local",
+			}
+			extensionSecret.Type = "Opaque"
+		})
+
+		It("should create the extension secret and extension BackupEntry if it doesn't exist yet", func() {
+			result, err := reconciler.Reconcile(ctx, request)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(Equal(reconcile.Result{}))
+
+			Expect(seedClient.Get(ctx, client.ObjectKeyFromObject(extensionSecret), extensionSecret)).To(Succeed())
+			Expect(extensionSecret.Data).To(BeEmpty())
+			Expect(extensionSecret.Labels).To(Equal(map[string]string{
+				"security.gardener.cloud/purpose":                   "workload-identity-token-requestor",
+				"workloadidentity.security.gardener.cloud/provider": "local",
+			}))
+			Expect(extensionSecret.Annotations).To(HaveKeyWithValue("workloadidentity.security.gardener.cloud/context-object", `{"kind":"BackupEntry","apiVersion":"core.gardener.cloud/v1beta1","name":"bar","namespace":"test","uid":""}`))
+			Expect(extensionSecret.Annotations).To(HaveKeyWithValue("workloadidentity.security.gardener.cloud/name", workloadIdentity.Name))
+			Expect(extensionSecret.Annotations).To(HaveKeyWithValue("workloadidentity.security.gardener.cloud/namespace", workloadIdentity.Namespace))
+			Expect(extensionSecret.Annotations).To(HaveKey("gardener.cloud/timestamp"))
+			Expect(extensionSecret.Type).To(BeEquivalentTo("Opaque"))
+
+			Expect(seedClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupEntry), extensionBackupEntry)).To(Succeed())
+			Expect(extensionBackupEntry.Spec).To(MatchFields(IgnoreExtras, Fields{
+				"DefaultSpec": MatchFields(IgnoreExtras, Fields{
+					"Type":           Equal(backupBucket.Spec.Provider.Type),
+					"ProviderConfig": Equal(backupBucket.Spec.ProviderConfig),
+				}),
+				"Region":                     Equal(backupBucket.Spec.Provider.Region),
+				"BackupBucketProviderStatus": Equal(backupBucket.Status.ProviderStatus),
+				"SecretRef": MatchFields(IgnoreExtras, Fields{
+					"Name":      Equal(extensionSecret.Name),
+					"Namespace": Equal(extensionSecret.Namespace),
+				}),
+			}))
+		})
+
+		It("should not reconcile the extension BackupEntry if the secret data or extension spec hasn't changed", func() {
```

Comment:
same here

Reply:
If you by "same here" mean it should be 
```suggestion
		It("should not reconcile the extension BackupEntry if the secret data or extension spec have not changed", func() {
```

I think it would be wrong. Or maybe you meant something else?

Reply:
Meant the same as https://github.com/gardener/gardener/pull/12321#discussion_r2170609377